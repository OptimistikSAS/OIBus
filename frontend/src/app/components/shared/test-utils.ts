import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
