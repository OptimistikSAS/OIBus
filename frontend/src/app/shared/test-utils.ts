import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
