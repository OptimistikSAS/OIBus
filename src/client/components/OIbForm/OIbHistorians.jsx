import React from 'react'
import PropTypes from 'prop-types'
import { Row, Col } from 'reactstrap'
import OIbTitle from './OIbTitle.jsx'
import Table from '../table/Table.jsx'

const OIbHistorians = ({ lastCompletedAt }) => {
  const renderTable = () => (
    <Table
      headers={['Type', 'Last Completed At', '']}
      rows={Object.keys(lastCompletedAt).map((key, i) => {
        const date = new Date(lastCompletedAt[key])
        return [
          {
            name: `key.${i}`,
            value: key,
          },
          {
            name: `value.${i}`,
            value: `${date.toLocaleDateString()} - ${date.toLocaleTimeString()} `,
          },
        ]
      })}
    />
  )

  return (
    <>
      <OIbTitle label="Historian support" />
      <Row>
        <Col md="6">
          {renderTable()}
        </Col>
      </Row>
    </>
  )
}

OIbHistorians.propTypes = { lastCompletedAt: PropTypes.object.isRequired }
OIbHistorians.defaultProps = {}

export default OIbHistorians
