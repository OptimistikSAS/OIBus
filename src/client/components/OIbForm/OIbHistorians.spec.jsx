import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

import OIbHistorians from './OIbHistorians.jsx'

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})
const lastCompletedAt = {
  name: 'OPCUA-HA',
  lastCompletedAt: {
    every10Seconds: 1615379954419,
    every1Min: 1615379954419,
  },
}

describe('OIbInteger', () => {
  test('check OIbInteger with value="a text"', () => {
    act(() => {
      ReactDOM.render(<OIbHistorians
        lastCompletedAt={lastCompletedAt}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
})
