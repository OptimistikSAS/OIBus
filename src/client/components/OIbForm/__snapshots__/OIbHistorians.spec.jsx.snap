// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OIbInteger check OIbInteger with value="a text" 1`] = `
<div>
  <div
    class="row"
  >
    <h5>
      Historian support
    </h5>
  </div>
  <div
    class="container-fluid"
  />
  <div
    class="row"
  >
    <div
      class="col-md-6"
    >
      <table
        class="table table-sm table-striped table-hover"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              Type
            </th>
            <th
              scope="col"
            >
              Last Completed At
            </th>
            <th
              scope="col"
            />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              name
            </td>
            <td>
              Invalid Date - Invalid Date 
            </td>
            <td />
          </tr>
          <tr>
            <td>
              lastCompletedAt
            </td>
            <td>
              Invalid Date - Invalid Date 
            </td>
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
